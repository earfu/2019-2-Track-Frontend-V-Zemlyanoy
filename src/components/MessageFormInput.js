import React from 'react';
import PropTypes from 'prop-types';
import chatDefaults from '../chatDefaults';

function MessageFormInput({ onChange, onKeyPress, onSubmit, value }) {
  return (
    <input
      type="text"
      className="message-form-input"
      placeholder={chatDefaults.messageInputText}
      onChange={onChange}
      value={value}
      onSubmit={onSubmit}
      onKeyPress={onKeyPress}
    />
  );
}

MessageFormInput.propTypes = {
  onKeyPress: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
};

export default MessageFormInput;
