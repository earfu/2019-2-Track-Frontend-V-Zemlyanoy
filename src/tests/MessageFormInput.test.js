import React from 'react';
import renderer from 'react-test-renderer';
import MessageFormInput from '../components/MessageFormInput';

it('renders correctly', () => {
  const component = renderer
    .create(
      <MessageFormInput
        value="Test message input text"
        onKeyPress={() => {}}
        onSubmit={() => {}}
        onChange={() => {}}
        onButtonClick={() => {}}
        onImageClear={() => {}}
        onImageLoad={() => {}}
        onAudioClear={() => {}}
        onAudioLoad={() => {}}
      />,
    )
    .toJSON();
  // empty function used; a warning here is fine, function calls are tested in Storybook
  expect(component).toMatchSnapshot();
});
