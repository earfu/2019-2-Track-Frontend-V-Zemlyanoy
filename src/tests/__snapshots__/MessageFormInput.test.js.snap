// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="message-sending-form"
>
  <form
    className="message-text-form"
  >
    <textarea
      className="message-form-input"
      cols={80}
      onChange={[Function]}
      onKeyPress={[Function]}
      onSubmit={[Function]}
      placeholder="Введите сообщение"
      type="text"
      value="Test message input text"
    />
    <button
      className="message-button"
      onClick={[Function]}
      type="submit"
    >
      Send message
    </button>
  </form>
  <form
    className="image-input-form"
  >
    <p
      className="image-input-text"
    >
      Image:
    </p>
    <input
      accept="image/*"
      id="image-input"
      onChange={[Function]}
      type="file"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </form>
  <form
    className="audio-input-form"
  >
    <p
      className="audio-input-text"
    >
      Audio:
    </p>
    <input
      accept="audio/*"
      id="audio-input"
      onChange={[Function]}
      type="file"
    />
    <button
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </form>
</div>
`;
